sdk/cloudmachine/Azure.CloudMachine/src/CloudMachineClientConfiguration.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Primitives;
+using System.Linq;
+
+namespace Azure.Core;
+
+/// <summary>
+/// extensions.
+/// </summary>
+public static class CloudMachineClientConfiguration
+{
+    /// <summary>
+    /// Adds a connection to the collection.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="connections"></param>
+    /// <returns></returns>
+    public static IConfigurationBuilder AddCloudMachineConnections(this IConfigurationBuilder builder, ConnectionCollection connections)
```

Comment:
should CloudMachineInfrastructure.AddFeature emit connections to the config?

Reply:
you mean instead of having the ConnectionCollection? If yes, then I don't think so. The config system thing is super slow in comparison and we retrieve connections often.

Reply:
I was thinking as a way to hydrate the ConnectionCollection - for example, after configure is run the first time clients wouldn't need to call AddFeature at runtime each time for extensible features.

Reply:
the connections configuration is not persisted. It's in memory. We can think about how to make it persistent, but it would be a separate PR.